Lib9c/Action/ActionObsoleteAttributeExtension.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections.Generic;
+using System.Linq;
+using Nekoyume.Action;
+
+namespace Nekoyume
+{
+    public static class ActionObsoleteAttributeExtension
+    {
+        public static bool IsObsolete(
+            this IEnumerable<ActionObsoleteAttribute> attrs,
+            Planet? planet,
+            long blockIndex
+        )
+        {
+            // it means that action had been obsoleted in the single planet era.
+            // in that case, check index in Odin only, rest should be obsoleted.
+            if (attrs.Count() == 1)
+            {
+                if (planet != Planet.Odin)
+                {
+                    return true;
+                }
+
+                // Comparison with ObsoleteIndex + 2 is intended to have backward compatibility with
+                // a bugged original implementation.
+                return attrs.Single().ObsoleteIndex + 2 <= blockIndex;
+            }
+
+            return attrs.Any(attr => attr.ObsoleteIndex <= blockIndex && attr.AffectedPlanet == planet);
```

Comment:
Does it need to check `ObsoleteIndex + 2 <= blockIndex` too when the `attr.AffectedPlanet` is `Planet.Odin`?

Reply:
AFIK, +2 offset was needed to aide previous bug and it doesn't be needed against new chains. am I understanding correctly? @greymistcube

Reply:
Yes, `Planet.Heimdall` and `Heimdall.Idun` will not need the +2 offset. But I think `Planet.Odin` needs +2 offset as I said. Odin is not a new chain.

Reply:
```suggestion
            return attrs.Any(attr => (attr.AffectedPlaent == Planet.Odin ? attr.ObsoleteIndex + 2 : attr.ObsoleteIndex) <= blockIndex && attr.AffectedPlanet == planet);
```

@moreal Ah, I see. you mean this?

Reply:
@longfin Ah yes!! That is what I mean.